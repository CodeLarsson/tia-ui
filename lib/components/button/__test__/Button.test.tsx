import React from 'react';
import { describe, expect, it, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import Button from '../Button';

describe('Button component', () => {
  it('Button should render correctly', () => {
    // Arrange
    render(<Button />);
    const button = screen.getByRole('button');

    // Assert
    expect(button).toBeInTheDocument();
  });

  it('Button should render primary class correctly', () => {
    // Arrange
    render(<Button primary={true} />);
    const button = screen.getByRole('button');

    // Assert
    expect(button.className).toContain('tia-bg-primary-500');
  });

  it('Button should render secondary class correctly', () => {
    // Arrange
    render(<Button primary={false} />);
    const button = screen.getByRole('button');

    // Assert
    expect(button.className).toContain('tia-bg-white');
  });

  it('Button should trigeger function on click', () => {
    // Arrange
    const mockFn = vi.fn();
    render(<Button onClick={mockFn} />);
    const button = screen.getByRole('button');

    // Act
    button.click();

    // Assert
    expect(mockFn).toHaveBeenCalled();
  });

  it('Button should match snapshot', () => {
    // Arrange
    const { container } = render(<Button />);

    //Assert
    expect(container).toMatchSnapshot();
  });
});
