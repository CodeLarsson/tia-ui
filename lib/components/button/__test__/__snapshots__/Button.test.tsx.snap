// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > Button should match snapshot 1`] = `
<div>
  <button
    class="tia-text-md tia-h-medium tia-px-4 tia-rounded-full tia-border tia-border-solid tia-bg-primary-500 tia-text-white tia-border-primary-600 tia-shadow-md tia-cursor-pointer hover:tia-bg-primary-600 hover:tia-shadow-lg transform transition-transform hover:tia-scale-105 "
    type="button"
  />
</div>
`;
